comments/serializers.py

```diff
@@ -66,6 +68,10 @@ def get_changed_my_mind(self, comment: Comment) -> dict[str, bool | int]:
     def get_text(self, value: Comment):
         return _("deleted") if value.is_soft_deleted else value.text
 
+    def get_edited_at(self, comment: Comment):
+        latest_diff = comment.comment_diffs.order_by('-created_at').first()
```

Comment:
This solution with produce [N+1 queries problem](https://docs.sentry.io/product/issues/issue-details/performance-issues/n-one-queries/).

> the value that gets set is very slightly after the created_at field for whatever reason so you can't even do a edited_at != created_at

I see what you mean. It’s likely that our `edited_at` field is automatically populated during question creation, but with a slight delay of a few microseconds from the created_at timestamp. This is expected since these operations execute sequentially.

Here are two possible solutions:
- Simple workaround (my fav): Treat questions as “non-edited” if edited_at is either null or within 1 second of created_at.
- More complex solution: Refactor the logic for edited_at assignment and create a custom data migration to normalize the edited_at field for existing records.

Reply:
hm I meant this as a reply: https://github.com/Metaculus/metaculus/pull/1221#discussion_r1826236448